pilot/pkg/xds/delta.go

```diff
@@ -393,21 +382,13 @@ func (s *DiscoveryServer) shouldRespondDelta(con *Connection, request *discovery
 	previousInfo.AlwaysRespond = false
 	con.proxy.Unlock()
 
-	oldAck := slices.EqualUnordered(previousResources, currentResources)
 	// Spontaneous DeltaDiscoveryRequests from the client.
 	// This can be done to dynamically add or remove elements from the tracked resource_names set.
-	// In this case response_nonce is empty.
-	newAck := request.ResponseNonce != ""
-	if newAck != oldAck {
```

Comment:
@howardjohn  I have another confusion， does the xds protocol gurantee a new spontaneous request definitely would cause previousResources!=currentResources. I cannot find that saying

Reply:
No, that is why we should keep the newAck if we are going to keep one. The protocol is clear that we should use responseNonce

Reply:
If no, then the current code would definitely panic when  assert, though i first reverted to original state

Reply:
it can never happen in envoy but can in general. note it's a debug assert only

Reply:
not debug, a error log. plus a panic here

Reply:
Lets remove it, cause many confusion to us